Improve translation of power/silo usage tooltip

dragunoff opened the pull request:
Improve translation of power/silo usage tooltip

- Fix an instance where "silo-usage" translation was used without arguments
- Use the same translation reference for the "Power usage"
- Make the ResourceBarWidget accept a cached transform with the tooltip text so it won't have to build the string itself
- Display an infinity symbol when the infinite power cheat is used
- Removes a magic number that is no longer used (>1000000 to check for
unlimited power)

![image](https://user-images.githubusercontent.com/1355810/190139542-a2d58da9-bded-4c9b-b702-10c4112945b7.png)

~~The FreeSans font update is here because the version we have currently (20090104) doesn't have the infinity symbol.~~

Fixes #20317

dragunoff committed:
Improve translation of power/silo usage tooltip

- Fix an instance where "silo-usage" translation was used without
arguments
- Use the same translation reference for the "Power usage"
- Make the ResourceBarWidget accept a cached transform with the tooltip
text
so it won't have to build the string itself
- Display an infinity symbol when the infinite power cheat is used
- Removes a magic number that is no longer used (>1000000 to check for
unlimited power)

abcdefg30 commented on OpenRA.Mods.Common/Widgets/Logic/Ingame/IngamePowerBarLogic.cs:
```diff
@@ -17,18 +17,30 @@ namespace OpenRA.Mods.Common.Widgets.Logic
-			powerBar.TooltipFormat = modData.Translation.GetString(PowerUsage) + ": {0}/{1}";
+			powerBar.TooltipTextCached = new CachedTransform<(float Current, float Capacity), string>(usage =>
+			{
+				var capacity = usage.Capacity < 0 ? modData.Translation.GetString(Unlimited) : usage.Capacity.ToString();
```
```suggestion
				var capacity = developerMode.UnlimitedPower ? modData.Translation.GetString(Unlimited) : usage.Capacity.ToString();
```
?

dragunoff commented:
I just noticed that the updated font looks quite different than the ancient version that we use.

Before
![Screenshot from 2022-09-14 14-53-41](https://user-images.githubusercontent.com/1355810/190148275-5d672f36-c48d-406b-9532-bd6c940c5e6f.png)

After
![Screenshot from 2022-09-14 14-52-37](https://user-images.githubusercontent.com/1355810/190148285-57672868-a795-4c67-a346-2b0952e3b5fe.png)

dragunoff commented:
If we can't find a solution to the font problem I'm fine with dropping the font changes and using a word instead of the infinity symbol:
![image](https://user-images.githubusercontent.com/1355810/190591417-1648262f-f83a-44c9-8f45-51dc45226e3c.png)

Mailaender commented on mods/common/languages/en.ftl:
```diff
@@ -202,11 +202,12 @@ exit-map-editor-confirm = Exit
 ## IngamePowerCounterLogic
-power-usage = Power Usage
+power-usage = Power Usage: { $used }/{ $capacity }
+infinity-symbol = ∞
```
Math is a universal language that doesn't need translation?

Mailaender commented on OpenRA.Mods.Common/Widgets/Logic/Ingame/IngamePowerCounterLogic.cs:
```diff
@@ -17,22 +17,34 @@ namespace OpenRA.Mods.Common.Widgets.Logic
-			power.GetText = () => powerManager.PowerProvided == 1000000 ? "∞" : powerManager.ExcessPower.ToString();
-			power.GetTooltipText = () => powerUsage + ": " + powerManager.PowerDrained.ToString() +
-				(powerManager.PowerProvided != 1000000 ? "/" + powerManager.PowerProvided.ToString() : "");
+			power.GetText = () => developerMode.UnlimitedPower ? modData.Translation.GetString(Unlimited) : powerManager.ExcessPower.ToString();
```
This gets translated on every render tick.

Mailaender commented on OpenRA.Mods.Common/Widgets/Logic/Ingame/IngamePowerCounterLogic.cs:
```diff
@@ -17,22 +17,34 @@ namespace OpenRA.Mods.Common.Widgets.Logic
+
+				return modData.Translation.GetString(
+					PowerUsage,
+					Translation.Arguments("used", powerManager.PowerDrained, "capacity", capacity));
```
This probably needs a `CachedTransform`.

Mailaender commented:
> I just noticed that the updated font looks quite different than the ancient version that we use.

Could be hinting problems. #3436

dragunoff commented:
![image](https://user-images.githubusercontent.com/1355810/193317711-83023309-8b7e-4c93-87e6-93e375868811.png)

Update: Addressed comments (https://github.com/OpenRA/OpenRA/pull/20295#discussion_r973607018, https://github.com/OpenRA/OpenRA/pull/20295#discussion_r973607283 and https://github.com/OpenRA/OpenRA/pull/20295#discussion_r973607342), dropped the font update and replaced "∞" with the word "Infinite". We can work on the font update in a dedicated future PR.

PunkPun commented on OpenRA.Mods.Common/Widgets/Logic/Ingame/IngamePowerBarLogic.cs:
```diff
@@ -17,18 +17,32 @@ namespace OpenRA.Mods.Common.Widgets.Logic
 		static readonly string PowerUsage = "power-usage";
 
+		[TranslationReference]
+		static readonly string Unlimited = "infinite-power";
```
```suggestion
		static readonly string Infinite = "infinite-power";
```

?

PunkPun commented on OpenRA.Mods.Common/Widgets/Logic/Ingame/IngamePowerCounterLogic.cs:
```diff
@@ -17,22 +17,39 @@ namespace OpenRA.Mods.Common.Widgets.Logic
 		static readonly string PowerUsage = "power-usage";
 
+		[TranslationReference]
+		static readonly string Unlimited = "infinite-power";
```
same here

Mailaender commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/dca33ad250cad23048b288df3f89d832f96f3d98...6a9ba415e2948b0d63d329371d4dab4d84f21961)
